File: android/src/main/kotlin/com/artahc/rx_method_channel/ObservableCallback.kt
package com.artahc.rx_method_channel

import com.google.gson.Gson
import com.google.gson.annotations.SerializedName

data class ObservableCallback(
    val requestId: Int,
    val value: Any?
) {
    fun toJson(): String {
        return Gson().toJson(this).toString()
    }
}

File: ios/Classes/ObservableCallback.swift
//
//  ObservableCallback.swift
//  rx_method_channel
//
//  Created by Komang Arta Wibawa on 31/01/23.
//

import Foundation

public struct ObservableCallback {
    let requestId: Int
    let value: Any?
    
    func toJson() throws -> String {
        let dict = [
            "requestId": requestId,
            "value": value
        ];
        
        do {
            let jsonObject = try JSONSerialization.data(withJSONObject: dict,  options: [])
            return String(data: jsonObject, encoding: .utf8)!
        } catch {
            throw ObservableCallbackError.parsingError
        }
    }
}

enum ObservableCallbackError: Error {
    case parsingError
}
